### PR TITLE
Add Labels from MR to Argocd AppProject 

### LINKED ISSUE #105: Support AppProject Labels
### What problem are you facing?

We are working on creating an ArgoCD project using the Crossplane provider. One of the key functionalities we rely on in ArgoCD is the usage of labels in the AppProject. These labels are crucial as they allow the AppProject to inherit default spec values from a global AppProject, providing us with a standardized setup and reducing manual configuration effort.

However, we've observed that the Crossplane provider's Managed Resource does not currently support adding these labels to the ArgoCD projects. This limitation hinders our ability to seamlessly integrate and manage our ArgoCD projects using Crossplane.

### How could Crossplane help solve your problem?

To address this challenge, it would be beneficial if provider argocd could enhance  to support Labels in AppProject: Add functionality in the Managed Resource to support defining and adding labels to the ArgoCD AppProject. This should be in line with how ArgoCD natively supports labels.

### PR DESCRIPTION
**Pull request by smileisak**
### Description of your changes

This pull request introduces the capability for the argocd provider to support defining and adding labels to the ArgoCD AppProject from the MR. Labels in ArgoCD play a crucial role as they allow AppProjects to inherit default spec values from a global AppProject.  [Documentation link](https://argo-cd.readthedocs.io/en/stable/user-guide/projects/#configuring-global-projects-v18)

Fixes #105 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

The code has been tested using a local setup where an ArgoCD project was created using the argocd provider. I verified:

- The labels are correctly added to the AppProject.
- There were no synchronization issues or delays.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/controller/projects/controller.go [hunk 1]
```diff
@@ -319,7 +319,7 @@ func generateCreateProjectOptions(p *v1alpha1.Project) *project.ProjectCreateReq
 	projectCreateRequest := &project.ProjectCreateRequest{
 		Project: &argocdv1alpha1.AppProject{
 			Spec:       projSpec,
-			ObjectMeta: metav1.ObjectMeta{Name: p.Name},
+			ObjectMeta: metav1.ObjectMeta{Name: p.Name, Labels: p.ObjectMeta.Labels},
```
Comment (MisterMX) [inline pkg/controller/projects/controller.go:322]: I think we should distinguish between MR labels and ones we want to add to the project.

There should be an additional property `spec.forProvider.labels` or `spec.forProvider.projectLabels` where the one can specifiy which to add without mixing them with the labels added to the MR (like Crossplane specific ones).

**Review by MisterMX — CHANGES_REQUESTED** [review]

File: pkg/controller/projects/controller.go (hunk 1 above)
Comment (smileisak) [reply pkg/controller/projects/controller.go:322]: I agree, I wanted to do it like this in the beginning but I did not want to add new fields in the forProvider. I'll make the update ASAP.

**Comment by smileisak** [comment]
@MisterMX can you take a look on it. I have made the change as requested.

**Review by MisterMX — APPROVED** [review]
LGTM. Thank you very much @smileisak!

**Pull request merged**